src/transformers/pipelines/visual_question_answering.py

```diff
@@ -116,11 +137,17 @@ def __call__(self, image: Union["Image.Image", str], question: str = None, **kwa
 
     def preprocess(self, inputs, padding=False, truncation=False, timeout=None):
         image = load_image(inputs["image"], timeout=timeout)
-        model_inputs = self.tokenizer(
-            inputs["question"], return_tensors=self.framework, padding=padding, truncation=truncation
-        )
-        image_features = self.image_processor(images=image, return_tensors=self.framework)
-        model_inputs.update(image_features)
+
+        if not self._is_instruct_blip:
+            model_inputs = self.tokenizer(
+                inputs["question"], return_tensors=self.framework, padding=padding, truncation=truncation
+            )
+            image_features = self.image_processor(images=image, return_tensors=self.framework)
+            model_inputs.update(image_features)
+        else:
+            # InstructBlip processes the image and text simultaneously:
+            model_inputs = self.processor(images=image, text=inputs["question"], return_tensors=self.framework)
```

Comment:
we do not use the processor classes in the pipelines explicitly (@Narsil could explain why).

hence we use the image_processor and tokenizer separately inside the pipeline, so for InstructBLIP it would look very similar to BLIP-2

Reply:
Ok, so you basically reimplement what's in `Blip2Processor.__call__` within `VisualQuestionAnsweringPipeline.preprocess` using the tokenizer and image_processor already available? And since InstructBlip passes the text encoding to the image_processor as well, I still need a switch to handle the inputs differently, it's just that I don't need to load the InstructBlipProcessor? I think I'll understand. I'll update the PR.

Reply:
I looked into this and see a complication. The Blip2Processor [only requires](https://github.com/huggingface/transformers/blob/5a73316bed96e9e022f9679dc8d1a7487f4721e1/src/transformers/models/blip_2/processing_blip_2.py#L45) a tokenizer and image_processor, both of which are available to the pipeline. However, the InstructBlipProcessor [uses](https://github.com/huggingface/transformers/blob/5a73316bed96e9e022f9679dc8d1a7487f4721e1/src/transformers/models/instructblip/processing_instructblip.py#L50) a tokenizer, image_processor, _and_ qformer_tokenizer; the latter is not available within the pipeline object. So how can we do a forward pass without loading InstructBlipProcessor?

Reply:
In that case, we will need to load the qformer_tokenizer inside the pipeline, only if the model_type is "instructblip"

Reply:
Thanks for clarifying. I'll look into your suggestion.

Yes, I'm curious about the decision not to use the Processor classes in pipelines. It seems like the alternative is a lot of work duplicating functionality and pushing implementation details (e.g. the existence of qformers) into the pipeline code instead of using the processor abstraction.

Reply:
@NielsRogge Can you please take another look? I've modified the implementation as you suggested. Thanks!